src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -13,6 +13,7 @@ namespace Microsoft.Net.Http.Headers
     public static class HeaderUtilities
     {
         private static readonly int _int64MaxStringLength = 20;
+        private static readonly int _qualityValueMaxCharCount = 5;      // See RFC7231 5.3.1
```

Comment:
I was unsure about being this strict in my last comment. I spoke with @Tratcher and we figured it's probably better to be a little lenient here so let's set this value to 10.